homeassistant/components/sensor/uber.py

```diff
@@ -0,0 +1,247 @@
+    for product_id, product in timeandpriceest.products.items():
+        if wanted_product_ids is not None and product_id in wanted_product_ids:
+            dev.append(UberSensor('time',
+                                  timeandpriceest,
+                                  product_id, product))
+            dev.append(UberSensor('price',
+                                  timeandpriceest,
+                                  product_id, product))
+        elif wanted_product_ids is None:
+            dev.append(UberSensor('time',
+                                  timeandpriceest,
+                                  product_id,
+                                  product))
+            dev.append(UberSensor('price',
+                                  timeandpriceest,
+                                  product_id,
+                                  product))
+    add_devices(dev)
+
+
+# pylint: disable=too-few-public-methods
+class UberSensor(Entity):
+    """Implementation of an Uber sensor."""
+
+    def __init__(self, sensorType, products, product_id, product):
+        """Initialize the Uber sensor."""
+        self.data = products
+        self._product_id = product_id
+        self._product = product
+        self._sensortype = sensorType
+        self._name = self._product.get('display_name') + " " + self._sensortype
+        if self._sensortype == "time":
+            self._unit_of_measurement = "min"
+            self._state = int(self._product.get(
+                'time_estimate_seconds', 0
+            ) / 60)
+        elif self._sensortype == "price":
+            if self._product.get('price_details').get('low_estimate') is None:
+                self._unit_of_measurement = self._product.get(
+                    'price_details'
+                ).get(
+                    'currency_code'
+                )
+                self._state = int(self._product.get(
+                    'price_details'
+                ).get(
+                    'minimum'
+                ))
+            else:
+                self._unit_of_measurement = self._product.get(
+                    'price_details').get('currency_code')
+                self._state = int(
+                    self._product.get('price_details').get('low_estimate'))
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        distance_key = 'Trip distance (in '+self._product.get(
+            'price_details').get('distance_unit')+'s)'
+        distance_val = self._product.get('distance')
+        if self._product.get(
```

Comment:
Wow, what are you doing here, so many enters!

You only have to use `get` if there might be a chance that a key does not exist. It will then return `None` (or the second argument that you pass into `get`).

To break a long line in multiple lines, you can use parentheses around the `if` statements or use a `\` to break a line but treat the final multi-lines as 1.

If you want to test if a key exists in a dictionary, you can use the `in` test.

```python
if 'distance_unit' not in self._product['price_details'] or \
   'distance' not in self._product):
    distance_key = …
```